### PR TITLE
LPD-29981 Remove duplicate escape usage

### PR DESCRIPTION
**Pull request by fortunatomaldonado**
https://liferay.atlassian.net/browse/LPD-29981
https://liferay.atlassian.net/browse/LPD-29981

The label was being escaped twice and resulting in incorrect labels. 

The double escape was introduced in https://liferay.atlassian.net/browse/LPS-185312 with this commit https://github.com/liferay-frontend/liferay-portal/pull/3578/files adding the escape to true.

I looked for aui:option with HtmlUtil.escape for the label and removed it so it would not get escaped twice.

For a case like this where it hits a lot of different components, how do we want to make the playwright test?
Should FEI be the ones making this change in the first place or should each team be responsible for their own tests?

Thank you for the help!

### CODE REVIEW

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by dsanz** [comment]
Hei @fortunatomaldonado testing wise, we shall check just the tag usage. Teams are responsible for their own tests and probably won't worry about this but rather making proper usage of the taglib

**Review by markocikos — APPROVED** [review]
@fortunatomaldonado LGTM! :sparkles:

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
Thank you @markocikos for reviewing!

I added a playwright test to check the taglib behavior.
Please let me know if there is anything that needs to be changed.
Thank you!

File: modules/test/playwright/tests/portal-web/taglib-aui-option/auiOptionEscape.spec.ts
Comment (markocikos) [inline modules/test/playwright/tests/portal-web/taglib-aui-option/auiOptionEscape.spec.ts:1]: This would set two serious precedents:
1. A test specific for a taglib, not for a feature. We blur the difference sometimes with a sample module, where we test features of the sample, but there is not in this case.
2. A test for AUI. AUI is something we generally want to work towards eliminating, not covering with tests.

I think I may have been wrong in green-lighting this, and Dani was right about letting product teams handle this. This feels like opening a Pandora's box. 

The tests, if we are adding them, should be added each in the module where the feature is. This makes sense, as the changes are on usages of taglib, not the taglib itself. So, for each of the 12 changes, a test on its respective page. Much of these may not exists, so we would be writing a lot of tests for product teams, in areas of their responsibilities. Do we want to do this? @dsanz

**Review by markocikos — CHANGES_REQUESTED** [review]
@fortunatomaldonado a potentially blocking issue inline

File: modules/test/playwright/tests/portal-web/taglib-aui-option/auiOptionEscape.spec.ts
Comment (dsanz) [reply modules/test/playwright/tests/portal-web/taglib-aui-option/auiOptionEscape.spec.ts:1]: Yep this case is somehow complicated as it combines uncommon factors:

- We're testing a taglib usage, not the taglib itself. This can be acceptable if we had a sample widget, but...
- We're testing a AUI tag, which we'd like to remove in the future, we don't want to invest much in this so coverage will evolve on a case-by-case basis
- Our change affects various usages, in the product

Given all this, I see acceptable to rename the current file `auiOptionEscape.spec.ts` and place it in the module that makes use of the taglib (`users-admin-web` most likely). Then, USM team (Commerce now, IIRC) would be informed about this in case they want to change anything in the test.

As there are other uncovered cases, we can notify affected teams so that they can consider copying the pattern in the test we created (which is enough to assert the absence of the issue as long as tag is used as intended). This way we provide a basic coverage yet allow others to enhance it if they decide to

Makes sense?

Comment (fortunatomaldonado) [reply modules/test/playwright/tests/portal-web/taglib-aui-option/auiOptionEscape.spec.ts:1]: Thank you @dsanz and @markocikos for checking this and providing steps to move forward.
I changed the playwright tests to user-admin-web.

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
I'll go ahead an forward this to Brian Chan and send another PR to Commerce to make them aware of the situation.
Thank you all for the help!

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
@dsanz and @markocikos, can I get verification that the ci:test is not related to this pull?
I don't believe it is just would like a second opinion.
Thank you!

File: modules/test/playwright/tests/users-admin-web/customFieldLabelEscape.spec.ts
Comment (markocikos) [inline modules/test/playwright/tests/users-admin-web/customFieldLabelEscape.spec.ts:1]: When adding a file or code to a product team, it's always a good idea to look at their structure and try to adapt to it. See how [their files](https://github.com/liferay/liferay-portal/tree/master/modules/test/playwright/tests/users-admin-web) are organized to match pages (`usersAndOrganizations.spec.ts`) or features (`editOrganization.spec.ts`). This file would not fit either. It should not be a file at all, since the scope is too small. It should be one of the tests within an existing page or feature file.

File: modules/test/playwright/tests/users-admin-web/customFieldLabelEscape.spec.ts
Comment (markocikos) [inline modules/test/playwright/tests/users-admin-web/customFieldLabelEscape.spec.ts:17]: Please see https://github.com/liferay-frontend/liferay-portal/pull/4313#discussion_r1687632458 in your other PR. The same logic applies here, almost all variables in this file are not named correctly.

**Review by markocikos — CHANGES_REQUESTED** [review]
@fortunatomaldonado Please see comments inline. In addition, I don't think we should forward this to Brian. You should either open a PR on the product team's fork, or ask for their review of this PR on Slack. cc @dsanz

File: modules/test/playwright/tests/users-admin-web/customFieldLabelEscape.spec.ts
Comment (fortunatomaldonado) [reply modules/test/playwright/tests/users-admin-web/customFieldLabelEscape.spec.ts:1]: Thank you for the insight on this. I'll make the necessary changes!

File: modules/test/playwright/tests/users-admin-web/customFieldLabelEscape.spec.ts
Comment (fortunatomaldonado) [reply modules/test/playwright/tests/users-admin-web/customFieldLabelEscape.spec.ts:17]: Will work on this, thank you!

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
Hello @markocikos,

I made the file changes. Let me know if this looks good enough to fwd to users-admin-web team, which I believe would be commerce at this time?
Thank you!

**Review by markocikos — APPROVED** [review]
@fortunatomaldonado LGTM! :sparkles:

**Comment by dsanz** [comment]
@fortunatomaldonado thos looks ready to merge right?

**Comment by fortunatomaldonado** [comment]
@dsanz, I have to remove the commerce modules as it looks like they are not affected.
https://github.com/liferay-commerce/liferay-portal/pull/5358#issuecomment-2389763030

**Comment by dsanz** [comment]
Thanks @fortunatomaldonado , as those PRs are also closed, is this one ready to merge? (it just touches the expando)

**Comment by fortunatomaldonado** [comment]
@dsanz, I will go ahead and close this PR as it has been merged here: https://github.com/liferay-core-infra/liferay-portal/pull/7503

**Pull request closed without merging**
